### PR TITLE
Fetch the test filename from .meta/config.json

### LINKED ISSUE #31: (improvement) Fetch the solution file and test file from .meta/config.json
Instead of building those filenames in run.sh, simply extract them from the exercise's config.

This will remove the need to first check if the test file is `*.bats` and if not using `*_test.sh`.

This will future-proof us if we want to change the solution from `*.sh` to `*.bash` or remove the extension entirely.

### PR DESCRIPTION
**Pull request by glennj**
Reads the test filename from the config file.
The logic to build the test filename from the slug still exists, "just in case (tm)".
Only some of the test data gets the config.json file to test both paths.

This change pulled in `jq` as a required utility, so I rewrote the results.json generation to use it.
This resulted in prettier json output, so all new extected_results.json.

I also upgraded the bats-core version to v1.5.0. Ubuntu 20.04 only provides v1.2.0, so the Dockerfile RUN instructions have been rewritten accordingly.

Resolved #31

### CODE REVIEW

File: bin/run.sh [hunk 1]
```diff
@@ -46,9 +51,14 @@ run_tests() {
...
-    # test scripts may be (old) xxxx_test.sh
-    [[ -f "$test_file" ]] || test_file="${slug//-/_}_test.sh"
+    local test_file
+    if [[ -f .meta/config.json ]]; then
+        test_file=$(jq -r '.files.test[0]' .meta/config.json)
+    else
+        test_file="${slug//-/_}.bats"
+        # test scripts may be (old) xxxx_test.sh
+        [[ -f "$test_file" ]] || test_file="${slug//-/_}_test.sh"
+    fi
```
Comment (glennj) [inline bin/run.sh:61]: This is where the .meta/config.json is consumed.

File: bin/run.sh [hunk 2]
```diff
@@ -174,78 +186,55 @@ error() {
...
+    local tests=("$@")
+    local opts=(
+        --null-input
+        --argjson version "$INTERFACE_VERSION"
+        --arg status "$status"
+        --jsonargs
+    )
+    jq  "${opts[@]}" \
+        '{version: $version, status: $status, tests: $ARGS.positional}' \
+        "${tests[@]}"
```
Comment (glennj) [inline bin/run.sh]: Here is `jq` in action to generate the results.out content.

File: bin/run.sh [hunk 3]
```diff
@@ -12,6 +13,10 @@
 # First runs the tests with bats and saves the TAP output,
 # then parses that TAP file to produce the JSON file.
 
+# the version of the test-runner interface:
+# https://github.com/exercism/docs/blob/main/building/tooling/test-runners/interface.md
```
Comment (kotp) [inline bin/run.sh]: Since this is something that will change…

File: bin/run.sh [hunk 4]
```diff
@@ -12,6 +13,10 @@
...
+INTERFACE_VERSION=2
```
Comment (kotp) [inline bin/run.sh:18]: … but this is a static entry, should we link to that file with the SHA link instead of the dynamic version (with `main`)?

**Review by kotp — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**